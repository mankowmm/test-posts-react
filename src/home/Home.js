import React, { Component } from 'react';
import { Jumbotron } from 'reactstrap';
import { Link, BrowserRouter } from 'react-router-dom';
import './Home.css';

class Home extends Component {

    render() {
        return (
          <div className="Home">
            <Jumbotron>
              <h1>Hello. This is example app created with ReactJS</h1>
              <p className="lead">
                <Link className="btn btn-primary" to="/posts">See Posts</Link>
              </p>
            </Jumbotron>
          </div>
        );
                
    }
}

export default Home;

