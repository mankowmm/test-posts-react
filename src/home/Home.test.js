import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';
import { Link } from 'react-router-dom';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

it('renders correctly', () => {
    const tree = renderer
        .create(<MemoryRouter><Home/></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe("Home - test suite", () => {

    let wrapper;
    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter><Home/></MemoryRouter>
        );
    });

    it('Should render required H1', () => {
        const h1 = wrapper.find('h1');
        expect(h1.text()).toBe('Hello. This is example app created with ReactJS');
    });

    it('Should render link', () => {
        const button = wrapper.find('a .btn-primary');
        expect(button.text()).toBe('See Posts');
        
    });
});
