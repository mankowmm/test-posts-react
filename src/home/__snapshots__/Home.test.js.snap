// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Home"
>
  <div
    className="jumbotron"
  >
    <h1>
      Hello. This is example app created with ReactJS
    </h1>
    <p
      className="lead"
    >
      <a
        className="btn btn-primary"
        href="/posts"
        onClick={[Function]}
      >
        See Posts
      </a>
    </p>
  </div>
</div>
`;
